File: backend/src/main/java/fr/scpit/backend/category/Category.java
package fr.scpit.backend.category;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "categories")
public class Category {
    @Id
    private String categoryId;
    private String categoryName;
    private String categoryImage;
}

File: frontend/lib/configuration/category.dart
class Category {
  final String categoryId;
  final String categoryName;
  final String categoryImage;

  Category({
    required this.categoryId,
    required this.categoryName,
    required this.categoryImage,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    final categoryId = json['categoryId'] as String? ?? '';
    final categoryName = json['categoryName'] as String? ?? '';
    final categoryImage = json['categoryImage'] as String? ?? '';

    return Category(
      categoryId: categoryId,
      categoryName: categoryName,
      categoryImage: categoryImage,
    );
  }

  Map<String, dynamic> toJson() => {
    'categoryId': categoryId,
    'categoryName': categoryName,
    'categoryImage': categoryImage,
  };

}
